File: contracts/models/Playlist.kt
package contracts.models

import kotlinx.serialization.Serializable

/**
 * Curated or algorithmic media playlist (Packup).
 */
@Serializable
data class Playlist(
    val id: String,                 // pack_01j7b...
    val ownerId: String,            // User ID
    val title: String,
    val description: String? = null,
    val isPrivate: Boolean = true,
    val itemCount: Int = 0,
    val coverArtUri: String? = null,
    val createdAt: String,          // ISO-8601 UTC
    val updatedAt: String           // ISO-8601 UTC
)

/**
 * Reference item in a playlist pointing strictly to a canonical [Media].
 */
@Serializable
data class PlaylistItem(
    val id: String,                 // itm_01j7b...
    val playlistId: String,
    val mediaId: String,            // Canonical Media ID
    val positionOrder: Int,
    val addedAt: String             // ISO-8601 UTC
)

File: contracts/models/Playlist.ts
export interface Playlist {
  id: string; // pack_01j7b...
  ownerId: string;
  title: string;
  description?: string | null;
  isPrivate: boolean;
  itemCount: number;
  coverArtUri?: string | null;
  createdAt: string; // ISO-8601 UTC
  updatedAt: string; // ISO-8601 UTC
}

export interface PlaylistItem {
  id: string; // itm_01j7b...
  playlistId: string;
  mediaId: string; // Canonical Media ID
  positionOrder: number;
  addedAt: string; // ISO-8601 UTC
}
